src/types/internal.types.ts

```diff
@@ -24,6 +25,9 @@ export interface ExecutionParams {
   step: Step
   statusManager: StatusManager
   settings: InternalExecutionSettings
+  customConfigCallback?: (
```

Comment:
Please, let's find a better self-descriptive name for it. This one doesn't tell us what it is for.